### PR TITLE
centralize imports, fix deprecation warnings and an error, bring up t…

### PR DESCRIPTION
**Pull request by sbromberger**
…o 0.4

### CODE REVIEW

File: src/parallel_matmul.jl [hunk 1]
```diff
@@ -3,14 +3,6 @@
...
 # implement load balancing for multiplication
 
-import
-    Base.A_mul_B, Base.At_mul_B, Base.Ac_mul_B, Base.A_mul_B!, Base.At_mul_B!, Base.Ac_mul_B!, 
-    Base.sdata, Base.size, Base.display
-
-export
-    SharedBilinearOperator, SharedSparseMatrixCSC, 
-    share, display, sdata, operator, nfilled, size,
-    A_mul_B, At_mul_B, Ac_mul_B, A_mul_B!, At_mul_B!, Ac_mul_B!, *
```
Comment (madeleineudell) [inline src/parallel_matmul.jl]: Looks like the centralized export missed a bunch of these functions...

**Comment by madeleineudell** [comment]
Thanks for the PR! Looks like you missed a few exports, though.

Did you test it in parallel on 0.4 and observe any parallel speedup? I haven't been able to install 0.4 yet, and noticed the amount of speedup depended quite heavily on the (changing) internals during the development of 0.3.

File: src/parallel_matmul.jl (hunk 1 above)
Comment (sbromberger) [inline src/parallel_matmul.jl]: If you're importing a function from Base you technically don't need to reexport that function.

**Comment by sbromberger** [comment]
So I needed parallel sparse matrices for a new branch of LightGraphs.jl, and I found that parallelizing betweenness centrality across 3 cores made the calculation almost 60% faster (560 seconds vs ~1350 not in parallel). I'm not using any of the multiplication but I did notice that issue #6 is causing some problems. I'm trying to fix that now.

Another issue I found in my own code is that since floating-point arithmetic is not associating (that is, `0.6 + 0.1 + 0.1 != 0.6 + 0.2`, lots of small errors tend to accumulate. I've found that in my FP-accumulation, after about 300k accums, I'm getting accuracy to about 5 decimal places. It seems to decrease as a function of graph size.

In any case, 5 decimal places is good enough for centrality measures on large graphs, so I'm happy with it. #6 is a puzzler. I'm wondering whether the shared arrays are really as parallel-write-safe as we assume. Might have to explore DArrays as an alternative.

File: src/parallel_matmul.jl (hunk 1 above)
Comment (madeleineudell) [inline src/parallel_matmul.jl]: Great! Then I've learned something new.

**Comment by madeleineudell** [comment]
I just updated the discussion of issue #6; the issue is (probably) that the `+=` operation is not thread safe. This means multiplication (but not transpose multiplication) of shared sparse matrices will not be thread safe. 

The way to fix it is to introduce a separate result vector for each core and to add them up in serial at the end.

**Comment by sbromberger** [comment]
I'm pretty sure that was the approach I took last night as I was mucking about. I'll have a look when I get back home.

**Comment by sbromberger** [comment]
I think my latest change to `src/parallel_matmul.jl` fixes #6; at least it results consistently in ~
```
julia> max_error = findmax(abs(v1-u1))
(1.4210854715202004e-14,12586)
```
which is close enough for government work :)

**Comment by madeleineudell** [comment]
That looks great. Thanks!

**Pull request merged**